### DISCUSSION TITLE
brand new install - no templates or themes, php errors (tseed and sqliteDB), rail fails

### DISCUSSION THREAD

**Discussion by jkiser99dev** [combostrap/combo#105] [General]
Hi, New user, brand new install.  (Windows11, PHP 8.2.12, Apache.)  I installed Dokuwiki and all recommended plugins (sqlite, move, etc).  After installing Combo all it did was mess up a minimal start page with a canned header/footer.  No templates or themes were installed.  In the Dokuwiki config->Basic->Template the only option is Dokuwiki (I thought bootstrap should be there). Under Combo settings I see the 5.0.1 stylesheets, but no themes are available.  The rail pops up, but no links work at all.
(Same result on Linux and PHP 8.1)

What am I missing??  I see no way to import templates or themes into Combo.

PHP warnings:
2024-06-06 00:32:10
E_WARNING: Undefined array key "tseed"
C:\localweb\dokuwiki\lib\plugins\combo\action\staticresource.php(152)
#0 C:\localweb\dokuwiki\lib\plugins\combo\action\staticresource.php(152): dokuwiki\ErrorHandler::errorHandler(2, 'Undefined array...', 'C:\\localweb\\dok...', 152)
#1 C:\localweb\dokuwiki\inc\Extension\EventHandler.php(80): action_plugin_combo_staticresource->handleSendFile(Object(dokuwiki\Extension\Event), Array)
#2 C:\localweb\dokuwiki\inc\Extension\Event.php(75): dokuwiki\Extension\EventHandler->process_event(Object(dokuwiki\Extension\Event), 'BEFORE')
#3 C:\localweb\dokuwiki\lib\exe\fetch.php(113): dokuwiki\Extension\Event->advise_before()
#4 {main}

PHP Depricated:
helper_plugin_sqlite::init() is deprecated. It was called from ComboStrap\Sqlite::createOrGetSqlite() in C:\localweb\dokuwiki\lib\plugins\combo\ComboStrap\Sqlite.php:121 dokuwiki\plugin\sqlite\SQLiteDB should be used instead!

**Comment by gerardnico** [comment]
Thanks for the feedback jkieser.

There is no theme to choose as combostrap comes with a template engine.

https://combostrap.com/theme/what-is-a-template-z9tl87hg

It just takes over the show actions of Dokuwiki.

All other actions are shown via the standard dokuwiki engine.

All the best

**Reply by jkiser99dev** [reply]
Thanks for the reply! My point is that this is a new, pristine install and there is no template engine now.  Step 2 in your link requires a "Metadata Manager" that is simply a link on the rail that does not work.  None of the rail links work on Windows or LInux so I can't create or edit pages either. (I don't use IOS or a Mac).  It also disabled the docuwiki rail version, so to create or edit a page all commands must be on the URL ?id=mypage&do=create,  Uninstalling Combo gets it all working again.

I use Brave and Firefox browsers.  I've searched the installation files for the templates in a quest for the holy grail, but there is nothing.

This is an effort to transition 7000 Mediawiki pages to something else and this was the solution folks liked best from the web info.

I hope that helps.

**Reply by gerardnico** [reply]
Men. That's frustrating.  Sorry.

My best guess is that Bootstrap does not load.
The railbar uses Bootstrap to show modals (ie the metadata  manager and all popup)

I'm a little bit in the dark as I don't have access to your installation.

If you go to the devtool (F12 on your page in your browser), do you see any error:
* in the dev console ?
* in the network tab ?

**Reply by gerardnico** [reply]
If you are in a private network. CDN may be blocked.
You can try to turn it off.
https://combostrap.com/performance/strap-cdn-m55ukl58#usecdn

**Reply by jkiser99dev** [reply]
Yes - you nailed it.  No bootstrap.  The CDN's fail and sometimes get a 404.  I reset the dokuwiki config to allow CDN in the network section, but neither option (cdnjs.com or code.jquery.com) makes any difference.  In the Combo config the CDN option was checked and I'm using bootstrap 5.0.1.    I've been avoiding the temptation to look at the code assuming other new users may run into the same issue.  If you need me to play/test something, I am an ok php/js/networking developer and I use bootstrap5 and modals in my own web apps.

The same errors happen on the web linux server.
Here is the console display: (sorry for the poor formatting)

- Refused to apply style from 'https://cdn.jsdelivr.net/dokuwiki//npm/bootstrap@5.0.1/dist/css/bootstrap.min.css' because its MIME type ('text/plain') is not a supported stylesheet MIME type, and strict MIME checking is enabled. :  doku.php:47               

       GET https://cdn.jsdelivr.net/dokuwiki//npm/@popperjs/core@2.9.2/dist/umd/popper.min.js net::ERR_ABORTED 404 (Not Found)
doku.php#dokuwiki__top:1 

Refused to execute script from 'https://cdn.jsdelivr.net/dokuwiki//npm/@popperjs/core@2.9.2/dist/umd/popper.min.js' because its MIME type ('text/plain') is not executable, and strict MIME type checking is enabled. : doku.php:48 
        
       GET https://cdn.jsdelivr.net/dokuwiki//npm/bootstrap@5.0.1/dist/js/bootstrap.min.js net::ERR_ABORTED 404 (Not Found)
doku.php#dokuwiki__top:1 

Refused to execute script from 'https://cdn.jsdelivr.net/dokuwiki//npm/bootstrap@5.0.1/dist/js/bootstrap.min.js' because its MIME type ('text/plain') is not executable, and strict MIME type checking is enabled. : doku.php:192 

Uncaught ReferenceError: bootstrap is not defined
    at Object.getDataNamespace (doku.php:192:13)
    at Object.enable (doku.php:216:38)
    at doku.php:242:20

getDataNamespace @ doku.php:192
enable @ doku.php:216
(anonymous) @ doku.php:242
load (async)
(anonymous) @ doku.php:241
doku.php:192 

Uncaught ReferenceError: bootstrap is not defined
    at Object.getDataNamespace (doku.php:192:13)
    at Object.enable (doku.php:216:38)
    at doku.php:248:20
getDataNamespace @ doku.php:192
enable @ doku.php:216
(anonymous) @ doku.php:248
load (async)
(anonymous) @ doku.php:247
fetch.php?drive=combo&fetcher=raw&media=snippet%3Ajs%3Acombo-search-box.js&tseed=1717415116:47 

Uncaught Error: Popper was not found  at SearchBox.getPopper (fetch.php?drive=combo&fetcher=raw&media=snippet%3Ajs%3Acombo-search-box.js&tseed=1717415116:47:19)
    at SearchBox.init (fetch.php?drive=combo&fetcher=raw&media=snippet%3Ajs%3Acombo-search-box.js&tseed=1717415116:68:40)
    at doku.php:372:10
getPopper @ fetch.php?drive=combo&fetcher=raw&media=snippet%3Ajs%3Acombo-search-box.js&tseed=1717415116:47
init @ fetch.php?drive=combo&fetcher=raw&media=snippet%3Ajs%3Acombo-search-box.js&tseed=1717415116:68
(anonymous) @ doku.php:372
load (async)
(anonymous) @ doku.php:341
doku.php#dokuwiki__top:1 

Error while trying to use the following icon from the Manifest: http://localhost/dokuwiki/lib/exe/fetch.php?media=wiki:logo.svg (Download error or resource isn't a valid image)

**Reply by gerardnico** [reply]
Hy Jkiser ! 

Good to know that you master all this. That's strange. You may have found a bug.

This is not good:
`https://cdn.jsdelivr.net/dokuwiki//npm/bootstrap@5.0.1/dist/css/bootstrap.min.css`
It should be
`https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/css/bootstrap.min.css`

**Reply by gerardnico** [reply]
I have pushed a fix that is live on Combostrap.
It's working and I have made some test unfortunately, I can test for now all edge cases.

If you update your combostrap installation by following this instructions, your problem should be gone.
https://combostrap.com/release/how-to-update-combo-or-a-third-party-plugin-in65c0ya#how_to_update_the_combo_plugin

**Reply by jkiser99dev** [reply]
YES!  That fixed the problem! THANK YOU!
New console errors follow but do not affect anything.  I did copy my new logo.svg file everywhere and it works 99%.  I'll try to figure out where a copy may be missing.

Uncaught TypeError: Cannot read properties of null (reading 'appendChild')
    at mediaListener (doku.php?id=start:299:21)
    at doku.php?id=start:309:5
mediaListener @ doku.php?id=start:299
(anonymous) @ doku.php?id=start:309
load (async)
(anonymous) @ doku.php?id=start:289

Error while trying to use the following icon from the Manifest: http://localhost/dokuwiki/lib/exe/fetch.php?media=wiki:logo.svg (Download error or resource isn't a valid image)

**Reply by gerardnico** [reply]
Cool. I have added a test so that it will not appear again.

The first error is an error on the Dokuwiki Javascript. I can't do anything about it unfortunately.

The second is about your logo. It seems that you have uploaded it.

There is a dedicated page here with all instructions:
https://combostrap.com/get-started/how-to-configure-the-favicons-mii2jsjq

This is what uses the [manifest](https://combostrap.com/app/strap-web-app-manifest-xrg4jl5d#schema)

All the best !

**Reply by jkiser99dev** [reply]
Ahhh, yes.  Because the media manager link  didn't work I had copied the files directly, bypassing the manifest update.

Thank you for being so responsive!  That is a tremendous selling point in moving this project forward with this plugin.  I've updated my systems and started customizing things.  Hopefully I won't bother you any more!!

**Reply by gerardnico** [reply]
If you can't navigate within the popovers on the railbar (ie slot manager and historical breadcrumb), I just pushed a fix.

For some reason, the dismiss on Bootstrap was kicking on some Browser before any link navigation.
I just took over.

Thanks for the feedback and all the best.
Bothering is learning.

Thanks
Nico

**Reply by jkiser99dev** [reply]
Thanks!  I'm doing the download now on both systems.

**Comment by gerardnico** [comment]
Do you work with IOS (Mac)

**Reply by gerardnico** [reply]
By the way, the php message are warnings, no error.

SQLite is there because I can't define SQLite as a dependency in dokuwiki and therefore I still use the last call to not break actual installation.

I'm looking to the first warning.

**Comment by gerardnico** [comment]
For the first warning, I resolved it. It's because php8 does not allow any key access on an array without default.
https://github.com/ComboStrap/combo/issues/107
